### ISSUE TITLE
Suggestion: use GPT itself to fix compilation bugs

### ISSUE THREAD

**Issue by oliveirabruno01** [keijiro/ScriptGeneratorTest#22]
> I pressed the "Run" button, resulting in compilation errors.
> In many cases, ChatGPT fails to implement your command correctly. You might get the correct one by repeatedly pressing the "Run" button.
> 
> (That's why I think this PoC is impractical...)

I know it may sounds stupid but have you tried using "ChatGPT" itself to resolve the errors instead of having the user run the prompt again? You could put a maximum amount of attempts.

**Comment by keijiro** [comment]
It would be a nice option. I'd like to give a try when I have time.

**Comment by bh679** [comment]
I added " - Use namespaces UnityEditor & UnityEngine.\n" +
to the WrapPrompt and it seems to help get ride of the most common issues I have seen.

Just a suggestion. Love what you've done here!

**Comment by AspireOne** [comment]
> I added " - Use namespaces UnityEditor & UnityEngine.\n" + to the WrapPrompt and it seems to help get ride of the most common issues I have seen.
> 
> Just a suggestion. Love what you've done here!

Yep, exactly. @oliveirabruno01 this is a problem with the output being inconsistent, because ChatGPT fails to follow the prompt exactly. It definitely is capable of it though, so the fix is improving the prompt.

**Comment by ConnerHua** [comment]
> 我在 WrapPrompt 中添加了“- Use namespaces UnityEditor & UnityEngine.\n” + ，它似乎有助于解决我见过的最常见问题。
> 
> 只是一个建议。喜欢你在这里所做的一切！

thanks！great！

**Comment by ali9d** [comment]
null reference accurs

**Comment by dGameMaker** [comment]
Just need to replace method with 3 lines lads:
```
void CreateScriptAsset(string code)
{
    ProjectWindowUtil.CreateAssetWithContent(TempFilePath, code);
    AssetDatabase.ImportAsset(TempFilePath);
    EditorUtility.FocusProjectWindow();
}
```
